### ISSUE TITLE
[Feature Request]: Generate multi-step plan for CTE/view

### ISSUE THREAD

**Issue by aunjgr** [matrixorigin/matrixone#10525]
### Is there an existing issue for the same feature request?

- [X] I have checked the existing issues.

### Is your feature request related to a problem?

MO treats all CTEs as inline subqueries, since there were no support for multi-step plan in pipeline.

After multi-step plans is supported, we should convert a proportion of CTEs to multi-step plans.

For example, TPC-H Q15 has a CTE which is referrenced twice and  takes over most of the query time. If we generates a 2-step plan, hopefully the time cost is reduced by half.

### Describe the feature you'd like

If a query referrences some CTE/view for multiple times, we should generate a multi-step plan to avoid the CTE/view being run multiple times. CTE/view which is referced only once should remain inlined.

Depends on #10096.

**Comment by aunjgr** [comment]
Won't do.

We should implement full-fledged CSE(common subexpression elimination), instead of this ad-hoc workaround. Need more thinking later.
